lib/dal/tests/integration_test/property_editor.rs

```diff
@@ -32,3 +33,261 @@ async fn assemble(ctx: &DalContext) {
+        Some("sushi".into()),
+        Some("tokyo".to_string()),
+    )
+    .await
+    .expect("four items in map");
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("baby back ribs".into()),
+        Some("jupiter".to_string()),
+    )
+    .await
+    .expect("five items in map");
+
+    // Grab the children for the array and check that they match what they should be
+    let parrot_names_child_ids =
+        AttributeValue::get_child_av_ids_for_ordered_parent(ctx, parrot_names_value_id)
+            .await
+            .expect("get the vec of child ids");
+    let parrot_names_third_item = AttributeValue::get_by_id(
+        ctx,
+        *parrot_names_child_ids
+            .get(2)
+            .expect("get the id for the third item"),
+    )
+    .await
+    .expect("get the third item in the array");
+    let parrot_names_third_item_value = parrot_names_third_item
+        .value(ctx)
+        .await
+        .expect("get the value for this array item");
+
+    // The value of the third item should be "jessica"
+    assert_eq!(parrot_names_third_item_value, Some("jessica".into()));
+
+    // Grab the children for the map and check that they match what they should be
+    let treasure_child_ids =
+        AttributeValue::get_child_av_ids_for_ordered_parent(ctx, treasure_map_value_id)
+            .await
+            .expect("get the vec of child ids");
+    let treasure_second_item = AttributeValue::get_by_id(
+        ctx,
+        *treasure_child_ids
+            .get(1)
+            .expect("get the id for the second item"),
+    )
+    .await
+    .expect("get the second item in the map");
+    let treasure_second_item_value = treasure_second_item
+        .value(ctx)
+        .await
+        .expect("get the value for this map item");
+
+    // The value of the second item should be "coxinha"
+    assert_eq!(treasure_second_item_value, Some("coxinha".into()));
+
+    let treasure_second_item_key = treasure_second_item
+        .key(ctx)
+        .await
+        .expect("get the key for this map item");
+
+    // The key of the second item should be "rio"
+    assert_eq!(treasure_second_item_key, Some("rio".to_string()));
+
+    // Check that there are five items in the array
+    assert_eq!(parrot_names_child_ids.len(), 5);
+
+    // Check that there are five items in the map
+    assert_eq!(treasure_child_ids.len(), 5);
+
+    // ======================================================
+    // Test removing items from the array
+    // ======================================================
+
+    // Remove an item from the array prop
+    AttributeValue::remove_by_id(ctx, parrot_names_third_item.id())
+        .await
+        .expect("remove the third item in parrot_names array");
+    let parrot_names_child_ids =
+        AttributeValue::get_child_av_ids_for_ordered_parent(ctx, parrot_names_value_id)
+            .await
+            .expect("get the vec of child ids");
```

Comment:
I'm wondering if we should instead use `PropertyEditorValues` since that is what the frontend will ask for.

Example: https://github.com/systeminit/si/blob/e18ec5484872bb5a63e13d326e32ab0f66410817/lib/dal/tests/integration_test/before_funcs.rs#L79-L84

The benefit to this: we won't be using lower level functions and will be reflecting more of SDF's common behavior. I don't think this is worth blocking the test over, but something to consider.

Reply:
Yeah I kept that in mind, but  sifting through the assembled object that sdf returns would be a world of pain, so for this interation I think this will suffice.

At some point I wanna add testing functions that parse both attribute values and the attribute form fields themselves into something more tree like, so that writing assertions for this data would be easier

Reply:
> Yeah I kept that in mind, but sifting through the assembled object that sdf returns would be a world of pain, so for this interation I think this will suffice.

It'll be a pain, but we will want it. I don't think we can avoid it. Plus, there are a few helper functions on the struct already to help out.

That being said, I can see why it's not worth it for this iteration.
I do think we need to consider it as a "next time" thing though.

> At some point I wanna add testing functions that parse both attribute values and the attribute form fields themselves into something more tree like, so that writing assertions for this data would be easier

I think `materialized_view` may be what you are looking for?